shared/test/unit/mocks/mock_navigator_moz_settings.js

```diff
@@ -64,6 +64,11 @@
     var req = {
       addEventListener: function(name, cb) {
         req['on' + name] = cb;
+      },
+      then: function(cb, eb) {
+        // error-handler `eb`is never used, since the mock never fails.
+        return Promise.resolve(0).then(cb); // set resolve to 0.
+
```

Comment:
For the then-able use case, I figured the developer wouldn't want to manually control when it fires – Since that's something a Promise deliberately hides away from you.